runtime/runtime_controller.h

```diff
@@ -22,13 +22,89 @@
+/// per shell. This object may only be created, used, and collected on the UI
+/// task runner. Window state queried by the root isolate is stored by this
+/// object. In cold-restart scenarios, the engine may collect this collect
+/// before installing a new runtime controller in its place. The Clone method
+/// may be used by the engine to copy the currently accumulated window state so
+/// it can be referenced by the new runtime controller.
+///
 class RuntimeController final : public WindowClient {
  public:
+  //----------------------------------------------------------------------------
+  /// @brief      Creates a new instance of a runtime controller. This is
+  ///             usually only done by the engine instance associated with the
+  ///             shell.
+  ///
+  /// @param      client                      The runtime delegate. This is
+  ///                                         usually the `Engine` instance.
+  /// @param      vm                          A reference to a running Dart VM.
+  ///                                         The runtime controller must be
+  ///                                         collected before the VM is
+  ///                                         destroyed (this order is
+  ///                                         guaranteed by the shell).
+  /// @param[in]  isolate_snapshot            The isolate snapshot used to start
+  ///                                         the root isolate managed by this
+  ///                                         runtime controller. The isolate
+  ///                                         must be transitioned into the
+  ///                                         running phase manually by the
+  ///                                         caller.
+  /// @param[in]  task_runners                The task runners used by the shell
+  ///                                         hosting this runtime controller.
+  ///                                         This may be used by the isolate to
+  ///                                         scheduled asynchronous texture
+  ///                                         uploads or post tasks to the
+  ///                                         platform task runner.
+  /// @param[in]  snapshot_delegate           The snapshot delegate used by the
+  ///                                         isolate to gather raster snapshots
+  ///                                         of Flutter view hierarchies.
+  /// @param[in]  io_manager                  The IO manager used by the isolate
+  ///                                         for asynchronous texture uploads.
+  /// @param[in]  unref_queue                 The unref queue used by the
+  ///                                         isolate to collect resources that
+  ///                                         may reference resources on the
+  ///                                         GPU.
+  /// @param[in]  image_decoder               The image decoder
```

Comment:
Might be helpful to explain why this is here and not on the io_manager, or how it's different.

Reply:
The IO manager does more than just aid in image decode. I think the difference will become clearer once I document that.